### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by aclave1** [kirbysayshi/vash#58]
Hello @kirbysayshi, are there any plans to allow for typescript in templates? I am currently in my senior compiler construction class, so I am able to help with this. Let me know what you think. I've taken a look into the code and can see two different paths to implementing this. 

1). modify the lexer,parser,etc to parse typescript and transform the template into typescript, then run it through tsc.
2). detect the typescript, run it through the typescript compiler, reinput it back into the template and allow vash to finish compilation.

What are your thoughts on typescript support, and my most likely naive idea of how to implement it?

**Comment by kirbysayshi** [comment]
I recently merged in a long-standing "beta" branch into master, so be sure you take a second look. The public interface hasn't changed, but under the hood the parser and AST are completely different.

I don't have any plans to include TS support, but it shouldn't be too difficult to create a vash-typescript library that reproduces what [vash.compile](https://github.com/kirbysayshi/vash/blob/c5e27d82adb1785f74c9ad28f7734ec12bd0722f/index.js#L46) does, but instead of calling `runtime.link` you'd first shell out the generated code to the TS compiler (can't tell if TS exports an interface that allows for programmatic compilation vs shell-based compilation), and then dump that into `runtime.link`.

If that doesn't work (and it might not due to more strict than necessary JS parsing), then perhaps something can be relaxed in the lib/parser.js.

**Comment by kirbysayshi** [comment]
@aclave1 Hey, I took a quick shot at it. Maybe it works? Not too familiar with typescript execution semantics, but it at least compiles! https://gist.github.com/kirbysayshi/66d3c26b2748c73715ae

Really hacky, but hopefully it helps you if you decide to pursue this further. I took the typescript example from https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#a-simple-transform-function.
